### PR TITLE
Parser: recover on unfinished record expr fields

### PR DESCRIPTION
**Pull request by auduchinok**
Adds recovery for various cases with unfinished record and anon record expressions:

```fsharp
{ F = }
```

```fsharp
{| F = |}
```

```fsharp
{| F |}
```

```fsharp
{| f().F |}
```

### CODE REVIEW

**Comment by auduchinok** [comment]
The tests also show the peculiar way of how some unfinished records are parsed today. Consider these cases:

```fsharp
{ F1 =
  F2 = 2 }
```

```fsharp
{ F1 = 1
  F2 =
  F3 = 3 }
```

I'd expect them to be parsed as 2 and 3 fields respectively, with error recovery applied for the fields with missing expressions. What happens instead is fields following the unfinished ones are parsed as `=` binary expressions (AST: [1](https://fsprojects.github.io/fantomas-tools/#/ast?data=N4KABGBEAmCmBmBLAdrAzpAXFSAacUiaAYmolmPAIYA2as%2BEkAxgPZwWTBjECMYAXgA6yCMQBMgsJIC%2BeApFgAPAA5Vk0CgBcATgFdYIGUA), [2](https://fsprojects.github.io/fantomas-tools/#/ast?data=N4KABGBEAmCmBmBLAdrAzpAXFSAacUiaAYmolmPAIYA2as%2BEkAxgPZwWTBjECMYAXjC8AOsgjEATILESAzILAKAvngKRYADwAOVZNAoAXAE4BXWCGVA)) or even sequences of them ([AST](https://fsprojects.github.io/fantomas-tools/#/ast?data=N4KABGBEAmCmBmBLAdrAzpAXFSAacUiaAYmolmPAIYA2as%2BEkAxgPZwWTBjECMYAXjC8AOsgjEATILESAzILBzZPACyL1AXzwFIsAB4AHKsmgUALgCcArrBCagA)).

I doubt that there's actual code written like this. We've seen this tree shape before and it made it much harder to analyze this unexpected structure in the tooling.

It's possible to change the structure into the more expected one, but it'd be a breaking change. To me it almost fits into the strict indentation changes in F# 8, and it shouldn't be hard to do. What do you think?

**Comment by vzarytovskii** [comment]
> I doubt that there's many actual code written like this.

Well, that's famous last words..."don't think it's going to break anyone".

**Review by T-Gro — APPROVED** [review]

**Review by psfinaki — APPROVED** [review]

File: tests/service/data/SyntaxTree/Expression/Record - Field 12.fs.bsl
Comment (kerams) [inline tests/service/data/SyntaxTree/Expression/Record - Field 12.fs.bsl:1]: No error here?

Comment (auduchinok) [reply tests/service/data/SyntaxTree/Expression/Record - Field 12.fs.bsl:1]: Yes, this is what my comment about a possible change about.

**Pull request merged**